### PR TITLE
Add ability to pass in custom Tip component

### PR DESCRIPTION
**Pull request by job13er**
NOTE: This is actually a change I'd like to submit to  `0.4.x` as I need it with `react@15`, but I didn't see a branch for that, so I thought I'd submit it here as a way to start the conversation. 

- Adds a new, optional `Tip` property which allows a custom `Tip` component to be used by `Popover`. 
- Adds documentation of new `Tip` property
- Adds a new select in the playground example to allow toggling between built-in and custom `Tip`

I didn't see much in the way of tests for `Popover` itself, so I wasn't sure what to do on that front. I'd be happy to add tests for this feature given a little guidance about how you'd like the tests written.

### CODE REVIEW

File: examples/playground/main.js [hunk 1]
```diff
@@ -23,8 +23,7 @@ const createOption = (type) => (
     key: type,
     value: type,
     children: type
-  })
-)
+  }))
```
Comment (job13er) [inline examples/playground/main.js]: Sorry, not sure how that got in there. I'll fix that.

**Comment by PlayMa256** [comment]
@job13er PR has conflicts, can you please fix them?

FYI:

- Set an [upstream](https://help.github.com/articles/syncing-a-fork/) remote.
- `git fetch upstream`
- `git rebase upstream/master`
- fix conflicts.
- commit again!

:)

**Comment by job13er** [comment]
I know it has conflicts, if you look at the PR description, it points out
that I don’t want to submit to PR to master, because I need it with react
15. If you can create a branch for the previous version that supports react
15, I’d Be happy to rebase and submit a pr to that branch.

**Comment by job13er** [comment]
On the other hand, if you don’t want to have both branches, that’s fine
too. I can close the PR and just use our fork.

**Pull request closed without merging**